src/Symfony/Component/Form/Extension/Core/Type/DateIntervalType.php

```diff
@@ -192,6 +195,21 @@ public function configureOptions(OptionsResolver $resolver)
             return array_fill_keys($timeParts, $placeholder);
         };
 
+        $labelsNormalizer = function (Options $options, array $labels) {
+            return array_replace(array(
+                'years' => null,
+                'months' => null,
+                'days' => null,
+                'weeks' => null,
+                'hours' => null,
+                'minutes' => null,
+                'seconds' => null,
+                'invert' => 'Negative interval',
+            ), array_filter($labels, function ($label) {
```

Comment:
I opted for this approach as it was my own preference, but expected negative feedbacks about it. Apparently it doesn't look that bad, so I'm satisfied 😅 .

Apart from perfs, IMHO segregating similar options is cluttering form types and make documentation more complicated and verbose no gain. Of course it depends of use cases and options, as it makes sense for most of them to be separated from each others in order to benefits of fine control over OptionResolver features like validation and normalization. But even for some of them, a feature like #18134 would be interesting (with an extra cost, though).

My only fear was about consistency with other form types which opted for segregation and autocompletions with the Symfony plugin for PHPStorm for instance.